--- gnutls-3.7.0.orig/lib/gnutls.pc.in	2021-05-13 07:23:29.522147091 +0800
+++ gnutls-3.7.0/lib/gnutls.pc.in	2021-05-13 07:26:53.503780866 +0800
@@ -19,6 +19,6 @@
 URL: https://www.gnutls.org/
 Version: @VERSION@
 Libs: -L${libdir} -lgnutls
-Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBATOMIC_LIBS@ @GNUTLS_LIBS_PRIVATE@
+Libs.private: -lintl @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBATOMIC_LIBS@ @GNUTLS_LIBS_PRIVATE@ -liconv
 @GNUTLS_REQUIRES_PRIVATE@
 Cflags: -I${includedir}
