diff --git a/mingw-w64-gnutls/0006-iconv-add-it-to-pkg-config.patch b/mingw-w64-gnutls/0006-iconv-add-it-to-pkg-config.patch
new file mode 100644
index 000000000..12f0b755e
--- /dev/null
+++ b/mingw-w64-gnutls/0006-iconv-add-it-to-pkg-config.patch
@@ -0,0 +1,11 @@
+diff -Naur gnutls-3.7.0.orig/lib/gnutls.pc.in gnutls-3.7.0/lib/gnutls.pc.in
+--- gnutls-3.7.0.orig/lib/gnutls.pc.in	2021-05-13 07:23:29.522147091 +0800
++++ gnutls-3.7.0/lib/gnutls.pc.in	2021-05-13 07:26:53.503780866 +0800
+@@ -19,6 +19,6 @@
+ URL: https://www.gnutls.org/
+ Version: @VERSION@
+ Libs: -L${libdir} -lgnutls
+-Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBATOMIC_LIBS@ @GNUTLS_LIBS_PRIVATE@
++Libs.private: @LIBINTL@ @LIBSOCKET@ @INET_PTON_LIB@ @LIBPTHREAD@ @LIB_SELECT@ @TSS_LIBS@ @GMP_LIBS@ @LIBUNISTRING@ @LIBATOMIC_LIBS@ @GNUTLS_LIBS_PRIVATE@ -liconv
+ @GNUTLS_REQUIRES_PRIVATE@
+ Cflags: -I${includedir}
diff --git a/mingw-w64-gnutls/PKGBUILD b/mingw-w64-gnutls/PKGBUILD
index 4d736a236..43b52a4b6 100644
--- a/mingw-w64-gnutls/PKGBUILD
+++ b/mingw-w64-gnutls/PKGBUILD
@@ -14,12 +14,10 @@ url="https://www.gnutls.org/"
 options=('!zipman')
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-gmp"
-         "${MINGW_PACKAGE_PREFIX}-libidn2"
+         "${MINGW_PACKAGE_PREFIX}-libiconv"
          "${MINGW_PACKAGE_PREFIX}-libsystre"
          "${MINGW_PACKAGE_PREFIX}-libtasn1"
          "${MINGW_PACKAGE_PREFIX}-nettle>=3.6"
-         "${MINGW_PACKAGE_PREFIX}-p11-kit>=0.23.1"
-         "${MINGW_PACKAGE_PREFIX}-libunistring"
          "${MINGW_PACKAGE_PREFIX}-zlib"
          #"${MINGW_PACKAGE_PREFIX}-unbound"
         )
@@ -30,13 +28,15 @@ source=(https://www.gnupg.org/ftp/gcrypt/gnutls/v${_pkgver%.*}/${_realname}-${pk
         0001-add-missing-define.patch
         0003-gnutls-fix-external-libtasn1-detection.patch
         0004-disable-broken-examples.patch
-        0005-remove-coverage-rules.patch)
+        0005-remove-coverage-rules.patch
+        0006-iconv-add-it-to-pkg-config.patch)
 sha256sums=('49e2a22691d252c9f24a9829b293a8f359095bc5a818351f05f1c0a5188a1df8'
             'SKIP'
             'fe1d9df924f72ec82972df328467591e49fe82feacebaf2605d37cc059c5429e'
             '6493f69e782d60fe04de4b0040987e99851c522d0baf2fe25d10b85b63e97863'
             'cbf9e59fb2432d4ac1edcb70734023e074b8058fa4c010a036a40de8449f67fb'
-            '10ea5a84459af8d848788cc8871decd3b3cdd542b829883ebe4f6ec28d60578f')
+            '10ea5a84459af8d848788cc8871decd3b3cdd542b829883ebe4f6ec28d60578f'
+            '4145d07e7793dfc2d1b222b5e80f771e42901cfd66b407efdeb81009cf8a19f6')
 validpgpkeys=('0424D4EE81A0E3D119C6F835EDA21E94B565716F')  # "Simon Josefsson <simon@josefsson.org>"
 validpgpkeys+=('1F42418905D8206AA754CCDC29EE58B996865171') # "Nikos Mavrogiannopoulos <nmav@gnutls.org>
 validpgpkeys+=('462225C3B46F34879FC8496CD605848ED7E69871') # "Daiki Ueno <ueno@unixuser.org>"
@@ -76,6 +76,9 @@ build() {
     --disable-guile \
     --disable-libdane \
     --disable-tests \
+    --without-p11-kit \
+    --with-included-unistring \
+    --without-idn \
     gl_cv_double_slash_root=yes
 
   make
