diff --git a/mingw-w64-gnutls/PKGBUILD b/mingw-w64-gnutls/PKGBUILD
index 4d736a236..a8f273bf4 100644
--- a/mingw-w64-gnutls/PKGBUILD
+++ b/mingw-w64-gnutls/PKGBUILD
@@ -14,12 +14,9 @@ url="https://www.gnutls.org/"
 options=('!zipman')
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-gmp"
-         "${MINGW_PACKAGE_PREFIX}-libidn2"
          "${MINGW_PACKAGE_PREFIX}-libsystre"
          "${MINGW_PACKAGE_PREFIX}-libtasn1"
          "${MINGW_PACKAGE_PREFIX}-nettle>=3.6"
-         "${MINGW_PACKAGE_PREFIX}-p11-kit>=0.23.1"
-         "${MINGW_PACKAGE_PREFIX}-libunistring"
          "${MINGW_PACKAGE_PREFIX}-zlib"
          #"${MINGW_PACKAGE_PREFIX}-unbound"
         )
@@ -76,6 +73,8 @@ build() {
     --disable-guile \
     --disable-libdane \
     --disable-tests \
+    --without-p11-kit \
+    --with-included-unistring \
     gl_cv_double_slash_root=yes
 
   make
