diff --git a/mingw-w64-gnutls/PKGBUILD b/mingw-w64-gnutls/PKGBUILD
index 4d736a236..275d786e3 100644
--- a/mingw-w64-gnutls/PKGBUILD
+++ b/mingw-w64-gnutls/PKGBUILD
@@ -18,7 +18,6 @@ depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-libsystre"
          "${MINGW_PACKAGE_PREFIX}-libtasn1"
          "${MINGW_PACKAGE_PREFIX}-nettle>=3.6"
-         "${MINGW_PACKAGE_PREFIX}-p11-kit>=0.23.1"
          "${MINGW_PACKAGE_PREFIX}-libunistring"
          "${MINGW_PACKAGE_PREFIX}-zlib"
          #"${MINGW_PACKAGE_PREFIX}-unbound"
@@ -65,7 +64,6 @@ build() {
     --host=${MINGW_CHOST} \
     --target=${MINGW_CHOST} \
     --enable-shared \
-    --enable-static \
     --enable-cxx \
     --enable-nls \
     --disable-rpath \
@@ -76,6 +74,7 @@ build() {
     --disable-guile \
     --disable-libdane \
     --disable-tests \
+    --without-p11-kit \
     gl_cv_double_slash_root=yes
 
   make
