diff --git a/mingw-w64-gnutls/PKGBUILD b/mingw-w64-gnutls/PKGBUILD
index 628c726b9..64bd90e95 100644
--- a/mingw-w64-gnutls/PKGBUILD
+++ b/mingw-w64-gnutls/PKGBUILD
@@ -14,12 +14,10 @@ url="https://www.gnutls.org/"
 options=('!zipman')
 depends=("${MINGW_PACKAGE_PREFIX}-gcc-libs"
          "${MINGW_PACKAGE_PREFIX}-gmp"
-         "${MINGW_PACKAGE_PREFIX}-libidn2"
+         "${MINGW_PACKAGE_PREFIX}-libiconv"
          "${MINGW_PACKAGE_PREFIX}-libsystre"
          "${MINGW_PACKAGE_PREFIX}-libtasn1"
          "${MINGW_PACKAGE_PREFIX}-nettle>=3.6"
-         "${MINGW_PACKAGE_PREFIX}-p11-kit>=0.23.1"
-         "${MINGW_PACKAGE_PREFIX}-libunistring"
          "${MINGW_PACKAGE_PREFIX}-zlib"
          #"${MINGW_PACKAGE_PREFIX}-unbound"
         )
@@ -29,12 +27,14 @@ makedepends=("${MINGW_PACKAGE_PREFIX}-gcc"
 source=(https://www.gnupg.org/ftp/gcrypt/gnutls/v${_pkgver%.*}/${_realname}-${pkgver}.tar.xz{,.sig}
         0003-gnutls-fix-external-libtasn1-detection.patch
         0004-disable-broken-examples.patch
-        0005-remove-coverage-rules.patch)
+        0005-remove-coverage-rules.patch
+        0006-iconv-add-it-to-pkg-config.patch)
 sha256sums=('3777d7963eca5e06eb315686163b7b3f5045e2baac5e54e038ace9835e5cac6f'
             'SKIP'
             '6493f69e782d60fe04de4b0040987e99851c522d0baf2fe25d10b85b63e97863'
             'cbf9e59fb2432d4ac1edcb70734023e074b8058fa4c010a036a40de8449f67fb'
-            '10ea5a84459af8d848788cc8871decd3b3cdd542b829883ebe4f6ec28d60578f')
+            '10ea5a84459af8d848788cc8871decd3b3cdd542b829883ebe4f6ec28d60578f'
+            '4145d07e7793dfc2d1b222b5e80f771e42901cfd66b407efdeb81009cf8a19f6')
 validpgpkeys=('0424D4EE81A0E3D119C6F835EDA21E94B565716F')  # "Simon Josefsson <simon@josefsson.org>"
 validpgpkeys+=('1F42418905D8206AA754CCDC29EE58B996865171') # "Nikos Mavrogiannopoulos <nmav@gnutls.org>
 validpgpkeys+=('462225C3B46F34879FC8496CD605848ED7E69871') # "Daiki Ueno <ueno@unixuser.org>"
@@ -44,6 +44,7 @@ prepare() {
   patch -p1 -i ${srcdir}/0003-gnutls-fix-external-libtasn1-detection.patch
   patch -p1 -i ${srcdir}/0004-disable-broken-examples.patch
   patch -p1 -i ${srcdir}/0005-remove-coverage-rules.patch
+  patch -p1 -i ${srcdir}/0006-iconv-add-it-to-pkg-config.patch
 
   WANT_AUTOMAKE=latest autoreconf -fiv -I m4
 }
@@ -73,6 +74,9 @@ build() {
     --disable-guile \
     --disable-libdane \
     --disable-tests \
+    --without-p11-kit \
+    --with-included-unistring \
+    --without-idn \
     gl_cv_double_slash_root=yes
 
   make
